osquery/core/init.cpp

```diff
@@ -217,6 +217,24 @@ const size_t kDatabaseRetryDelay{200};
 std::function<void()> Initializer::shutdown_{nullptr};
 RecursiveMutex Initializer::shutdown_mutex_;
 
+namespace {
+
+void initWorkDirectories() {
+  if (!FLAGS_disable_database) {
+    auto const recursive = true;
+    auto const ignore_existence = true;
+    auto const status = createDirectory(
+        boost::filesystem::path(FLAGS_database_path).parent_path(),
+        recursive,
+        ignore_existence);
+    if (!status.ok()) {
+      LOG(WARNING) << "Could not initialize db directory " << status.what();
```

Comment:
I suppose it's more like an error. As afterwards things will crash if the folder could not be created